common/djangoapps/track/middleware.py

```diff
@@ -137,6 +138,7 @@ def enter_request_context(self, request):
             'session': self.get_session_key(request),
             'user_id': self.get_user_primary_key(request),
             'username': self.get_username(request),
+            'ip': get_ip(request),
```

Comment:
This will return `None` when there is no IP header set.  You are right that technically this is a departure from the current implementation which returns `''` in that case.  I can update the code to keep it consistent (though the risk of the base case `REMOTE_ADDR` being `None` is low/non-existent)